src/module.ts

```diff
@@ -286,3 +289,15 @@ declare module '@nuxt/schema' {
     'i18n:extend-messages': (messages: LocaleMessages<DefineLocaleMessage>[], localeCodes: string[]) => Promise<void>
   }
 }
+
+declare module '#app' {
+  interface RuntimeNuxtHooks {
+    'i18n:beforeLocaleSwitch': <Context = unknown>(
+      params: LocaleSwitch & {
+        initialSetup: boolean
+        context: Context
+      }
+    ) => MaybePromise<void>
```

Comment:
```suggestion
    ) => MaybePromise<string | void>
```

Reply:
See docs:
https://v8.i18n.nuxtjs.org/guide/callbacks#onbeforelanguageswitch
>Returns: string or nothing

Reply:
Instead of return, it may need to be something like `nuxtApp.$i18n.locale = 'en'`

Reply:
This won't work with `callHook` it seems!